### PR TITLE
Renamed and reapportioned yamls for letkf task

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
Renames/moves common yamls from letkf subdir and rearranges contents to serve for tools common to SOCA

Another step toward resolving https://github.com/NOAA-EMC/GDASApp/issues/1091, paired with g-w 
https://github.com/NOAA-EMC/global-workflow/pull/2635

### CODE REVIEW

File: parm/soca/soca_ens_bkg_stage.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+######################################
+mkdir:
+{% for mem in range(1, NMEM_ENS + 1) %}
+- "{{ DATA }}/ens/mem{{ '%03d' % mem }}"
+{% endfor %}
+######################################
+# copy ensemble background files
+######################################
+copy:
+{% for mem in range(1, NMEM_ENS + 1) %}
```
Comment (guillaumevernieres) [inline parm/soca/soca_ens_bkg_stage.yaml.j2:17]: hard to read, indent the loops

File: parm/soca/soca_fix_stage.yaml.j2 [hunk 2]
```diff
@@ -1,3 +1,5 @@
+mkdir:
+- "{{ DATA }}/INPUT"
 ######################################
```
Comment (guillaumevernieres) [inline parm/soca/soca_fix_stage.yaml.j2:4]: Can you add a `TODO` here specifying that this is resolution dependent ... I need to address this on my side too.

**Review by guillaumevernieres — DISMISSED** [review]
Good enough for now. But do address the cosmetic comment and add a reminder that the yaml for staging the fix files is resolution dependent.

File: parm/soca/soca_ens_bkg_stage.yaml.j2 (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply parm/soca/soca_ens_bkg_stage.yaml.j2:17]: done

File: parm/soca/soca_fix_stage.yaml.j2 (hunk 2 above)
Comment (AndrewEichmann-NOAA) [reply parm/soca/soca_fix_stage.yaml.j2:4]: done

**Comment by AndrewEichmann-NOAA** [comment]
Includes workaround described in https://github.com/NOAA-EMC/GDASApp/issues/1213

**Review by guillaumevernieres — APPROVED** [review]

**Review by guillaumevernieres — APPROVED** [review]
thanks @AndrewEichmann-NOAA

**Pull request merged**